test/framework/src/main/java/org/elasticsearch/test/AbstractSearchCancellationTestCase.java

```diff
@@ -281,7 +281,9 @@ public void onIndexModule(IndexModule indexModule) {
             indexModule.addSearchOperationListener(new SearchOperationListener() {
                 @Override
                 public void onNewReaderContext(ReaderContext c) {
-                    runOnNewReaderContext.get().accept(c);
+                    if (runOnNewReaderContext.get() != null) {
+                        runOnNewReaderContext.get().accept(c);
+                    }
```

Comment:
@JVerwolf take a gander, I noticed that this could throw NPEs as the plugin isn't always initialized in the tests.